src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java

```diff
@@ -371,7 +371,7 @@ private void importFromGcs(String projectId, TableReference tableRef, @Nullable
       }
 
       if (JobInfo.WriteDisposition.WRITE_TRUNCATE
-        .equals(JobInfo.WriteDisposition.valueOf(writeDisposition))) {
+        .equals(JobInfo.WriteDisposition.valueOf(writeDisposition)) && allowSchemaRelaxation) {
 
         Table table = bigQueryHelper.getTable(tableRef);
         List<TableFieldSchema> tableFieldSchemas = Optional.ofNullable(table)
```

Comment:
The table should be always present in this case. You don't need to use Optional here.

Reply:
@supriyapatil-google Done